helix-term/src/commands/typed.rs

```diff
@@ -913,6 +913,29 @@ fn setting(
     Ok(())
 }
 
+/// Change the language of the current buffer at runtime.
+fn language(
+    cx: &mut compositor::Context,
+    args: &[Cow<str>],
+    _event: PromptEvent,
+) -> anyhow::Result<()> {
+    if args.len() != 1 {
+        anyhow::bail!("Bad arguments. Usage: `:set-language language`");
+    }
+
+    let doc = doc_mut!(cx.editor);
+    let loader = Some(cx.editor.syn_loader.clone());
+    let config = cx.editor.syn_loader.language_configs().find_map(|config| {
```

Comment:
Rather than a custom implementation, set_language2 already takes a scope name and looks up the appropriate language. See the discussion on https://github.com/helix-editor/helix/pull/1787

Reply:
Maybe a `set_language3` that takes the name of a lang? doing `:lang scope.rust` isn't intuitive IMO.

Reply:
`set_language2(format!("scope.{}", language),...)`

Reply:
If that's the approach taken, then the fuzzy finding for the languages should be changed to look through the scopes rather than the language names. There are a few langs where the names and scope endings aren't 1-to-1.

Reply:
I didn't use `scope` because the prefixes don't seem to follow the `source.<lang>` format all the time and the language id doesn't always match with suffix either. 
Here's a list of those that don't match with language id:
 - protobuf source.proto
 - c-sharp source.csharp
 - javascript source.js
 - typescript source.ts
 - html text.html.basic
 - latex source.tex
 - ocaml-interface source.ocaml.interface
 - racket source.rkt
 - comment scope.comment
 - llvm-mir source.llvm_mir
 - llvm-mir-yaml source.yaml
 - markdown source.md
 - git-commit git.commitmsg
 - git-diff source.diff
 - git-rebase source.gitrebase
 - git-config source.gitconfig
 - solidity source.sol